sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -307,6 +335,14 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
 
   private def timeZoneId = DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone)
 
+  protected override def getDefaultTreePatternBits: BitSet = {
```

Comment:
Let's override `lazy val treePatternBits` instead.

Reply:
Done.

Reply:
@sigmod I mean `treePatternBits` instead of the method `getDefaultTreePatternBits`

Reply:
Done. Thanks @gengliangwang!